odoo/addons/base/models/ir_module.py

```diff
@@ -250,8 +250,6 @@ def _get_icon_image(self):
             if module.icon:
                 path_parts = module.icon.split('/')
                 path = modules.get_module_resource(path_parts[1], *path_parts[2:])
-            elif module.id:
-                path = modules.module.get_module_icon(module.name)
```

Comment:
> for the compute function _get_icon_image else condition is removed as it seems not needed and previously if code execution enter the else it throws exception

"It seems not needed" is not a valid rationale to get rid of this if-branch. We cannot just remove a piece of working code when it reveals a bug some where else. Can you verify that this piece of code is indeed not needed anymore?